docs/articles/Vector-Indexes.md

```diff
@@ -0,0 +1,326 @@
+
+#### Inverted File (IVF)
+IVF works similarly to LSH and creates groups of data. But rather than hashing it, it uses clustering techniques. The techniques can vary depending on the implementation, but simpler techniques may use K-means clustering and then use the cluster centroids as a reference for query vectors. The query vector is then compared with only its associated data cluster to improve efficiency. IVF has a few variations, each improving the storage and retrieval efficiency.
+
+##### IVFFLAT
+The flat variation clusters the vectors and creates centroids but stores each vector as it is. It performs no additional processing beyond the clustering, and a brute-force approach is required to search through any given cluster.
+
+##### IVFPQ
+Inverted File Product Quantization (IVFPQ) improves the vector storage cost by quantizing the data. It begins by clustering the data points, but before storing the vectors, it quantizes the data. The algorithm follows the following steps:
+1. Divide the vector space into clusters.
+2. Break vectors within the cluster into smaller chunks. For example, if the original vector consists of 12 elements, it can be broken into 3 chunks, each with 4 elements.
+3. Quantize the chunks (4-dimensional vectors) into bits using product quantization.
+
+At search time, the same algorithm is applied to the query vector. It is first associated with a cluster by calculating the distance between the query and the cluster centroids. Then, it is broken into chunks and quantized. The encoded sub-vectors are compared with the encoded data within the cluster using distance metrics. This storage method has various benefits, such as it saves on space, and since we’re comparing smaller vectors, it improves vector search times.
+##### IVFSQ
+Inverted File Scalar Quantization (IVFSQ) is similar to IVFPQ but uses a simpler quantization algorithm. It begins by clustering the data points and then quantizing each vector separately. The quantization algorithm quantizes each vector dimension. The first step is to create bins to map the vector elements to. Each element is mapped to a bin, and depending on the bin range, the floating point number is converted to a scalar integer.
+
+Suppose we have a vector x=[1.2,3.5,−0.7,2.1]. As it has 4 dimensions, we will define 4 quantization bins such as the following:
+- Bin 0: [−1.0,0.0)[-1.0, 0.0)[−1.0,0.0)
+- Bin 1: [0.0,1.0)[0.0, 1.0)[0.0,1.0)
+- Bin 2: [1.0,2.0)[1.0, 2.0)[1.0,2.0)
+- Bin 3: [2.0,4.0)[2.0, 4.0)[2.0,4.0)
+
+Each vector element will be distributed into a bin as follows: 
+- For x1=1.2: It falls into Bin 2 [1.0,2.0), so it is quantized to 2.
+- For x2=3.5: It falls into Bin 3 [2.0,4.0), so it is quantized to 3.
+- For x3=−0.7: It falls into Bin 0 [−1.0,0.0), so it is quantized to 0.
+- For x4=2.1: It falls into Bin 3 [2.0,4.0), so it is quantized to 3.
+
```

Comment:
is it correct to replicate each of these bins three times? or is the correct version the following?

- Bin 0: [−1.0,0.0)
- Bin 1: [0.0,1.0)
- Bin 2: [1.0,2.0)
- Bin 3: [2.0,4.0)